### PR TITLE
SW-3275 End-to-end test

### PR DESCRIPTION
**Pull request by devkokov**

### CODE REVIEW

File: elasticsearch/client.go [hunk 1]
```diff
@@ -25,7 +26,9 @@ func NewClient(logger *log.Logger) (*Client, error) {
 	return &Client{logger: logger}, nil
 }
 
-func (c Client) Index(i Indexable) error {
+func (c Client) Index(i Indexable) *IndexResult {
+	c.logger.Printf("Indexing %s ID %d", reflect.TypeOf(i).PkgPath(), i.GetId())
```
Comment (denialanderror) [inline elasticsearch/client.go:30]: Assuming you are wanting to print the type here, you can do the following instead of using reflection to get the package path:

`c.logger.Printf("Indexing %T ID %d", i, i.GetId())`

File: go.mod [hunk 2]
```diff
@@ -6,4 +6,5 @@ require (
 	github.com/aws/aws-sdk-go v1.32.3
```
Comment (denialanderror) [inline go.mod:6]: We should ensure the rest of these dependencies are also up to date as the aws sdk and mux are both behind.

File: person/person.go [hunk 3]
```diff
@@ -1,10 +1,15 @@
 package person
 
 type Person struct {
+	Id        int
 	FirstName string
 	LastName  string
 }
 
+func (p Person) GetId() int {
```
Comment (denialanderror) [inline person/person.go:9]: According to [this](https://golang.org/doc/effective_go#Getters), getters in Go should leave out the "get" part, e.g. `Id()`

File: test.DockerFile [hunk 4]
```diff
@@ -0,0 +1,3 @@
+FROM golang:1.14
```
Comment (denialanderror) [inline test.DockerFile:1]: This can be bumped to 1.16

File: wait-for-it/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FROM alpine:3.10
```
Comment (denialanderror) [inline wait-for-it/Dockerfile:1]: Looks like this is a few versions behind also

**Review by denialanderror — CHANGES_REQUESTED** [review]
Mostly style and convention comments  but I do think we should make sure all the versions are up to date before merging.

I'm not a massive fan of the shortened variable names as it makes it a bit hard to follow but I don't know if that is a Go convention or not.

File: elasticsearch/client.go (hunk 1 above)
Comment (devkokov) [reply elasticsearch/client.go:30]: Good point! Thanks

File: go.mod (hunk 2 above)
Comment (devkokov) [reply go.mod:6]: All dependencies updated

File: person/person.go (hunk 3 above)
Comment (devkokov) [reply person/person.go:9]: Sorted

File: test.DockerFile (hunk 4 above)
Comment (devkokov) [reply test.DockerFile:1]: Sorted

File: wait-for-it/Dockerfile (hunk 5 above)
Comment (devkokov) [reply wait-for-it/Dockerfile:1]: Updated

**Review by denialanderror — APPROVED** [review]
